packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-config.ts

```diff
@@ -1,46 +1,109 @@
- * Note: CloudFormation does not currently support creating custom ECS configs outside
- * of using a custom resource. You can import custom deployment config created outside the
- * CDK or via a custom resource with {@link EcsDeploymentConfig#fromEcsDeploymentConfigName}.
  */
 export interface IEcsDeploymentConfig {
+  /** @attribute */
   readonly deploymentConfigName: string;
+
+  /** @attribute */
   readonly deploymentConfigArn: string;
 }
 
+/**
+ * Construction properties of {@link EcsDeploymentConfig}.
+ */
+export interface EcsDeploymentConfigProps {
+  /**
+   * The physical, human-readable name of the Deployment Configuration.
+   * @default - automatically generated name
+   */
+  readonly deploymentConfigName?: string;
+
+  /**
+   * The configuration that specifies how traffic is shifted from the 'blue'
+   * target group to the 'green' target group during a deployment.
+   * @default AllAtOnce
+   */
+  readonly trafficRoutingConfig?: ITrafficRoutingConfig;
+}
+
 /**
  * A custom Deployment Configuration for an ECS Deployment Group.
  *
- * Note: This class currently stands as namespaced container of the default configurations
- * until CloudFormation supports custom ECS Deployment Configs. Until then it is closed
- * (private constructor) and does not extend {@link Construct}
- *
  * @resource AWS::CodeDeploy::DeploymentConfig
  */
-export class EcsDeploymentConfig {
+export class EcsDeploymentConfig extends Resource implements IEcsDeploymentConfig {
   public static readonly ALL_AT_ONCE = deploymentConfig('CodeDeployDefault.ECSAllAtOnce');
+  public static readonly LINEAR_10PERCENT_EVERY_1MINUTES = deploymentConfig('CodeDeployDefault.ECSLinear10PercentEvery1Minutes');
+  public static readonly LINEAR_10PERCENT_EVERY_3MINUTES = deploymentConfig('CodeDeployDefault.ECSLinear10PercentEvery3Minutes');
+  public static readonly CANARY_10PERCENT_5MINUTES = deploymentConfig('CodeDeployDefault.ECSCanary10Percent5Minutes');
+  public static readonly CANARY_10PERCENT_15MINUTES = deploymentConfig('CodeDeployDefault.ECSCanary10Percent15Minutes');
 
   /**
    * Import a custom Deployment Configuration for an ECS Deployment Group defined outside the CDK.
    *
-   * @param _scope the parent Construct for this new Construct
-   * @param _id the logical ID of this new Construct
+   * @param scope the parent Construct for this new Construct
+   * @param id the logical ID of this new Construct
    * @param ecsDeploymentConfigName the name of the referenced custom Deployment Configuration
    * @returns a Construct representing a reference to an existing custom Deployment Configuration
    */
-  public static fromEcsDeploymentConfigName(_scope: Construct, _id: string, ecsDeploymentConfigName: string): IEcsDeploymentConfig {
+  public static fromEcsDeploymentConfigName(scope: Construct, id: string, ecsDeploymentConfigName: string): IEcsDeploymentConfig {
+    ignore(scope);
+    ignore(id);
     return deploymentConfig(ecsDeploymentConfigName);
```

Comment:
We can do that for `fromEcsDeploymentConfigName` method, but we still need the `deploymentConfig` method for the static configs like `ALL_AT_ONCE`.  They don't have any stack/scope context.

Reply:
Yeah that's fine to keep it for the static methods. For `fromEcsDeploymentConfigName` using
`Stack.formatArn` would grab the values (region, account, etc) from the stack instead of using the
generic intrinsic values.